src/Upbit/Upbit.jl

```diff
@@ -0,0 +1,135 @@
+module Upbit
+
+export UpbitCommonQuery,
+    UpbitPublicQuery,
+    UpbitAccessQuery,
+    UpbitPrivateQuery,
+    UpbitAPIError,
+    UpbitClient,
+    UpbitData
+
+using Serde
+using Dates, NanoDates, Base64, Nettle
+using UUIDs, JSONWebTokens
+
+using ..CryptoAPIs
+using ..CryptoAPIs: Maybe,  AbstractAPIsError, AbstractAPIsData, AbstractAPIsQuery, AbstractAPIsClient
+
+abstract type UpbitData <: AbstractAPIsData end
+abstract type UpbitCommonQuery <: AbstractAPIsQuery end
+abstract type UpbitPublicQuery <: UpbitCommonQuery end
+abstract type UpbitAccessQuery <: UpbitCommonQuery end
+abstract type UpbitPrivateQuery <: UpbitCommonQuery end
+
+"""
+    UpbitClient <: AbstractAPIsClient
+
+Client info.
+
+## Required fields
+- `base_url::String`: Base URL for the client. 
+
+## Optional fields
+- `public_key::String`: Public key for authentication.
+- `secret_key::String`: Secret key for authentication.
+- `interface::String`: Interface for the client.
+- `proxy::String`: Proxy information for the client.
+- `account_name::String`: Account name associated with the client.
+- `description::String`: Description of the client.
+"""
+Base.@kwdef struct UpbitClient <: AbstractAPIsClient
+    base_url::String
+    public_key::Maybe{String} = nothing
+    secret_key::Maybe{String} = nothing
+    interface::Maybe{String} = nothing
+    proxy::Maybe{String} = nothing
+    account_name::Maybe{String} = nothing
+    description::Maybe{String} = nothing
+end
+
+struct UpbitAPIsErrorMsg
+    name::Int64
+    message::String
+end
+
+"""
+    UpbitAPIError{T} <: AbstractAPIsError
+
+Exception thrown when an API method fails with code `T`.
+
+## Required fields in UpbitAPIsErrorMsg
+- `name::Int64`: Error code.
+- `message::String`: Error message.
+
+## Required fields
+- `error::UpbitAPIsErrorMsg`: Error struct.
+"""
+struct UpbitAPIError{T} <: AbstractAPIsError
+    error::UpbitAPIsErrorMsg
+
+    function UpbitAPIError(error::UpbitAPIsErrorMsg)
+        return new{error.name}(error)
+    end
+end
+
+CryptoAPIs.error_type(::UpbitClient) = UpbitAPIError
+
+function Base.show(io::IO, e::UpbitAPIError)
+    return print(io, "name = ", "\"", e.error.name, "\"", ", ", "msg = ", "\"", e.error.message, "\"")
+end
+
+struct UpbitUndefError <: AbstractAPIsError
+    e::Exception
+    msg::String
+end
+
+function CryptoAPIs.request_sign!(::UpbitClient, query::Q, ::String)::Q where {Q<:UpbitPublicQuery}
+    return query
+end
+
+function CryptoAPIs.request_sign!(client::UpbitClient, query::Q, ::String)::Q where {Q<:UpbitPrivateQuery}
```

Comment:
@gryumova, can you add at least one private method so that we can check the signature?

Reply:
e.g. `v1/status/wallet`